### DISCUSSION TITLE
Should the app focus on non-readable elements?

### DISCUSSION THREAD

**Discussion by Irineu333** [Irineu333/SpeakTouch#77] [Question]
At present, **Speak Touch**, unlike other screen readers such as **Talkback** and **Jieshou**, does not focus on non-legible elements, even if they are clickable or focusable.

This approach was chosen because when focusing on an element with no content or child to be read, there is nothing to present to the user. In the case of Talkback, for example, it says "no label" for clickable elements and remains silent for focusable elements.

This left me with doubts because important elements of the interface may not have descriptions of their content. This is particularly concerning for clickable elements, so perhaps we should focus on all clickable elements, at least, and only notify the user with "no content" when there is no data to be read.

**Comment by Irineu333** [comment]
A solution for these elements without descriptions would be support for custom identifiers. Through a file, provide descriptions of elements based on the element's ID. I think Talkback has this functionality, along with icon identification.

**Comment by PatrykMis** [comment]
I often use this function in TalkBack. Personally I've turned on setting to speak Ids of unlabeled elements and this often helps a lot. Nothing is worse than the button which you even don't know it exists on the screen, so all clickable elements should be listed and announced regardless if they are properly labeled or not.

**Reply by Irineu333** [reply]
I understand, so you suggest the "speak ids" option as a solution. This feature is usually optional, so it depends on the settings screen [^1] under development.

How does this functionality behave in applications made with JetPack Compose?

[^1]: #39

**Reply by Irineu333** [reply]
What do you think of the “custom identifiers” functionality? Talkback has it but I haven't tested it. I believe it works by associating descriptions with the element's id (if it has an id).

**Reply by PatrykMis** [reply]
I think it's a good solution, especially in rare cases when a button doesn't have proper description and also no proper ID. Screen readers says just "button" then.

"Custom identifiers" is also good feature, especially if app has a lot of improperly labeled controlls and the user does not want to remember the role or meening each time. Personally I almost don't use that feature, as for me buttons which have IDs like `IcNotificationBtn` are pretty descriptive but I know some people who always describing that kind of controlls in screen reader options, either TalkBack on Android or VoiceOver on iOS.

**Reply by Irineu333** [reply]
I understand, so there are three issues:
1. Focus on non-legible buttons [^1]
2. Speak ids option [^2]
3. Custom identifiers [^3]

[^1]: #88 
[^2]: #89
[^3]: #90
